tests/phpunit/integration/Core/Assets/AssetsTest.php

```diff
@@ -255,10 +255,14 @@ public function test_base_data__product_base_paths__hidden_post_type() {
 	/**
 	 * @dataProvider data_product_base_paths
 	 */
-	public function test_base_data__product_base_paths( $post_type_args, $expected ) {
+	public function test_base_data__product_base_paths( $args, $expected ) {
 		$this->enable_feature( 'userInput' );
 		$this->set_permalink_structure( '/%postname%/' );
-		register_post_type( 'product', $post_type_args );
+		register_post_type( 'product', $args['post_type_args'] );
+
+		if ( ! empty( $args['product_permastruct'] ) ) {
```

Comment:
The tests as they stood wouldn't have provided coverage for the permalink/regexp substitution, so I added a bit of coverage.